languages/go/bitwarden_client.go

```diff
@@ -0,0 +1,67 @@
+package sdk
+
+import (
+	"encoding/json"
+
+	"github.com/bitwarden/sdk/languages/go/internal/cinterface"
+)
+
+type BitwardenClient struct {
+	client        cinterface.ClientPointer
+	lib           cinterface.BitwardenLibrary
+	commandRunner CommandRunnerInterface
+	Projects      ProjectsInterface
+	Secrets       SecretsInterface
+}
+
+func NewBitwardenClient(apiURL *string, identityURL *string) (*BitwardenClient, error) {
+	deviceType := DeviceType("SDK")
+	userAgent := "Bitwarden GOLANG-SDK"
+	clientSettings := ClientSettings{
+		APIURL:      apiURL,
+		IdentityURL: identityURL,
+		UserAgent:   &userAgent,
+		DeviceType:  &deviceType,
+	}
+
+	settingsJSON, err := json.Marshal(clientSettings)
+	if err != nil {
+		return nil, err
+	}
+
+	lib := cinterface.NewBitwardenLibrary()
+	client, err := lib.Init(string(settingsJSON))
+	if err != nil {
+		return nil, err
+	}
+	runner := NewCommandRunner(client, lib)
+
+	return &BitwardenClient{
+		lib:           lib,
+		client:        client,
+		commandRunner: runner,
+		Projects:      NewProjects(runner),
+		Secrets:       NewSecrets(runner),
+	}, nil
+}
+
+func (c *BitwardenClient) AccessTokenLogin(accessToken string) error {
+	req := AccessTokenLoginRequest{AccessToken: accessToken}
+	command := Command{AccessTokenLogin: &req}
+
+	responseStr, err := c.commandRunner.RunCommand(command)
+	if err != nil {
+		return err
+	}
+
+	var response ResponseForAPIKeyLoginResponse
+	if err := json.Unmarshal([]byte(responseStr), &response); err != nil {
+		return err
+	}
+
+	return nil
```

Comment:
We need to do checkSuccessAndError here as well:
```go
func (c *BitwardenClient) AccessTokenLogin(accessToken string) (*ResponseForAPIKeyLoginResponse, error) {
	req := AccessTokenLoginRequest{AccessToken: accessToken}
	command := Command{AccessTokenLogin: &req}

	responseStr, err := c.commandRunner.RunCommand(command)
	if err != nil {
		return nil, err
	}

	var response ResponseForAPIKeyLoginResponse
	if err = checkSuccessAndError(responseStr, &response); err != nil {
		return nil, err
	}

	return &response, nil
}
```

Reply:
do we want response retuned ?

Reply:
Good point, I didn't see that was changed to not return a response.

We only need to return an error yeah, but at least we should check that success is true in the response like we do with checkSuccessAndError, maybe like this?

```go
func (c *BitwardenClient) AccessTokenLogin(accessToken string) error {
	req := AccessTokenLoginRequest{AccessToken: accessToken}
	command := Command{AccessTokenLogin: &req}

	responseStr, err := c.commandRunner.RunCommand(command)
	if err != nil {
		return err
	}

	var response APIKeyLoginResponse
	if err = checkSuccessAndError(responseStr, &response); err != nil {
		return err
	}

	return nil
}
```

Reply:
APIKeyLoginResponse or ResponseForAPIKeyLoginResponse? I would say ResponseForAPIKeyLoginResponse

Reply:
Changed

Reply:
changed to APIKeyLoginResponse